pkg/reconciler/persistentvolumeclaim/storagesource.go

```diff
@@ -131,33 +133,35 @@ func getCandidateSourceFromBackupList(ctx context.Context, backupList apiv1.Back
 		contextLogger.Debug("found a backup that is a valid storage source candidate",
 			"backupName", backup.Name)
 
-		result := &StorageSource{
-			DataSource: corev1.TypedLocalObjectReference{
-				APIGroup: ptr.To(volumesnapshot.GroupName),
-				Kind:     "VolumeSnapshot",
-				Name:     GetName(backup.Name, utils.PVCRolePgData),
-			},
+		return getCandidateSourceFromBackup(backup)
+	}
+
+	return nil
+}
+
+func getCandidateSourceFromBackup(backup *apiv1.Backup) *StorageSource {
```

Comment:
in future, we should also return an error if there is no match for the element.type

Reply:
Yes. Especially when we add the tablespaces, which have one more degree of freedom (the tablespace name)